src/OrleansProviders/Streams/Memory/MemoryMessageBody.cs

```diff
@@ -29,16 +30,28 @@ public interface IMemoryMessageBodySerializer
     /// Default IMemoryMessageBodySerializer
     /// </summary>
     [Serializable]
-    public class DefaultMemoryMessageBodySerializer : IMemoryMessageBodySerializer
+    public class DefaultMemoryMessageBodySerializer : IMemoryMessageBodySerializer, IOnDeserialized
```

Comment:
Understood.  I'd also prefer a solution that didn't alter the public surface, as this will be a breaking change, but I'm more open to breaking changes in the streaming system as that is still one of the newer systems which is, imo, still in flux.  We can also probably count the number of users that implemented custom IBatchContainers on one hand...